warp/backend.go

```diff
@@ -16,27 +16,28 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
-var _ Backend = &backend{}
-
 const batchSize = ethdb.IdealBatchSize
 
+var _ Backend = &backend{}
+
 // Backend tracks signature-eligible warp messages and provides an interface to fetch them.
 // The backend is also used to query for warp message signatures by the signature request handler.
 type Backend interface {
-	// AddMessage signs [unsignedMessage] and adds it to the warp backend database
+	// AddMessage stores [unsignedMessage] and this node's signature over it.
```

Comment:
We only store the message, not the signature, so that if the node's BLS signer changes after a restart it will create a signature that matches the up to date BLS Signer.

Reply:
updated comment

Reply:
This comment still reads a bit funny to me. We store the signature in the cache, but but saying that we store the message and the signature over it and then clarifying the signature handling after seems a bit messy. Would you mind doing another pass?